portal/src/main/webapp/WEB-INF/jsp/visualize.jsp

```diff
@@ -85,47 +85,52 @@
 
                 if (values != null && values.length >0)
                 {
-                    for (int i=0; i<values.length; i++)
-                    {
+                    for (int i=0; i<values.length; i++) {
+                    	
+                    	Boolean genesetQuery = false;
+		               	if (localGeneSetList.length() > 0 && localGeneList.length() == 0) {
+                    		genesetQuery = true;
+                    	}
+                    	
                         String currentValue = values[i].trim();
 
-                        if (currentValue.contains("mutation") && !disabledTabs.contains("mutations"))
+                        if ((currentValue.contains("mutation") && !disabledTabs.contains("mutations")) && !genesetQuery)
```

Comment:
well, `disabledTabs` and `genesetQuery` are just pointing to different things (the tabs that are disabled and whether we have gene sets in the query or not), so I would keep it like this. How do you suggest to update `disableTabs`?